Package.swift

```diff
@@ -74,6 +74,65 @@ let frameworksToChecksum = [
     "AWSUserPoolsSignIn": "e3ec5ec0c7a506c7305541099b5417deeeab0d9d3d09da6abf4ce4bfab2b3c4f"
 ]
 
+
+extension Target.Dependency {
+    // Framework dependencies present in the SDK
+    static let awsCore: Self = .target(name: "AWSCore")
+    static let awsAuthCore: Self = .target(name: "AWSAuthCore")
+    static let awsCognitoIdentityProviderASF: Self = .target(name: "AWSCognitoIdentityProviderASF")
+    static let awsCognitoIdentityProvider: Self = .target(name: "AWSCognitoIdentityProvider")
+}
+
+let depdenencyMap: [String: [Target.Dependency]] = [
+    "AWSAPIGateway": [.awsCore],
+    "AWSAppleSignIn": [.awsCore, .awsAuthCore],
+    "AWSAuthCore": [.awsCore],
+    "AWSAuthUI": [.awsCore, .awsAuthCore],
+    "AWSAutoScaling": [.awsCore],
+    "AWSChimeSDKIdentity": [.awsCore],
+    "AWSChimeSDKMessaging": [.awsCore],
+    "AWSCloudWatch": [.awsCore],
+    "AWSCognitoAuth": [.awsCore, .awsCognitoIdentityProviderASF],
+    "AWSCognitoIdentityProvider": [.awsCore, .awsCognitoIdentityProviderASF],
+    "AWSCognitoIdentityProviderASF": [.awsCore],
+    "AWSComprehend": [.awsCore],
+    "AWSConnect": [.awsCore],
+    "AWSConnectParticipant": [.awsCore],
+    "AWSCore": [],
+    "AWSDynamoDB": [.awsCore],
+    "AWSEC2": [.awsCore],
+    "AWSElasticLoadBalancing": [.awsCore],
+    "AWSFacebookSignIn": [.awsCore, .awsAuthCore],
+    "AWSGoogleSignIn": [.awsCore, .awsAuthCore],
+    "AWSIoT": [.awsCore],
+    "AWSKMS": [.awsCore],
+    "AWSKinesis": [.awsCore],
+    "AWSKinesisVideo": [.awsCore],
+    "AWSKinesisVideoArchivedMedia": [.awsCore],
+    "AWSKinesisVideoSignaling": [.awsCore],
+    "AWSLambda": [.awsCore],
+    "AWSLex": [.awsCore],
+    "AWSLocationXCF": [.awsCore],
+    "AWSLogs": [.awsCore],
+    "AWSMachineLearning": [.awsCore],
+    "AWSMobileClientXCF": [.awsAuthCore, .awsCognitoIdentityProvider],
+    "AWSPinpoint": [.awsCore],
+    "AWSPolly": [.awsCore],
+    "AWSRekognition": [.awsCore],
+    "AWSS3": [.awsCore],
+    "AWSSES": [.awsCore],
+    "AWSSNS": [.awsCore],
+    "AWSSQS": [.awsCore],
+    "AWSSageMakerRuntime": [.awsCore],
+    "AWSSimpleDB": [.awsCore],
+    "AWSTextract": [.awsCore],
+    "AWSTranscribe": [.awsCore],
+    "AWSTranscribeStreaming": [.awsCore],
+    "AWSTranslate": [.awsCore],
+    "AWSUserPoolsSignIn": [.awsCognitoIdentityProvider, .awsAuthCore, .awsCore]
```

Comment:
Do we need to add .awsCore here? Because awsAuthCore already depend on .awsCore.

Same for few others above which has both .awsCore, .awsAuthCore

Reply:
Yes, we do actually need to. The dependencies listed are each framework's **direct** dependencies. Omitting them leads to a build failre. 
This is following the same format as currently used in our podspecs.